### ISSUE TITLE
Input max length of 71 instead of 72 bytes.

### ISSUE THREAD

**Issue by mathieutu** [patrickfav/bcrypt#43]
Hi, 
Thanks for your work.
This issue to raise a point about the character limit you considered for password, and the way the null terminator byte is handled.

It is following:
- #42
- #22
- #27
- #13
- https://github.com/patrickfav/bcrypt/pull/28
- https://github.com/patrickfav/bcrypt/pull/23

From what I've understood from my readings, a password must be >= 72 bytes, and it should contain a null terminator byte since `$2a$`.
In your implementation, you consider this null byte as part of the 72, so put a hard limit of 71 bytes.

Like others before us, we're trying to make compatible this package to other implementations because we're using the password in several apps in other languages. However, all our passwords are prehashed, and their max length is **exactly 72 bytes**.

It worked for years with other implementations, but break with yours, because of the Strict Strategy error.
However, when using the pass-through strategy, or setting the limit to 72, everything is working great. We also tried to use the custom `VERSION_2Y_NO_NULL_TERMINATOR` to avoid this 71 limitation. However, if the generated passwords work for the 72 bytes inputs, they don't for smaller ones.

The [OpenBSD original implementation](https://github.com/openbsd/src/blob/6c1aad9dfe4cce499a6aa0dec4bd783c4505783b/lib/libc/crypt/bcrypt.c#L125-L128) itself, adds the null terminator byte to the 72 byte limit:
```c
	case 'b':
		/* strlen() returns a size_t, but the function calls
		 * below result in implicit casts to a narrower integer
		 * type, so cap key_len at the actual maximum supported
		 * length here to avoid integer wraparound */
		key_len = strlen(key);
		if (key_len > 72)
			key_len = 72;
		key_len++; /* include the NUL */
		break;
```

Some examples of implementations that work with 72 bytes inputs and add a null terminator byte at the end:
- [Python](https://github.com/pyca/bcrypt/blob/acf9f1c558a59cce50004c2a053ba7ed77c9ff9b/src/bcrypt/__init__.py#L82)
- [Rust](https://github.com/Keats/rust-bcrypt/blob/b6af5e5311598aee52d0bcfac1e22e13b1d10946/src/bcrypt.rs#L17)
- [Ruby](https://github.com/bcrypt-ruby/bcrypt-ruby/blob/e402d692b1bfd2623e9cb414afb2bbe190e9e500/lib/bcrypt/engine.rb#L17)
- [Node JS](https://github.com/kelektiv/node.bcrypt.js/blob/11d2ddd185c163314bd91754c5803002d929b4ff/src/bcrypt.cc#L224)
- [PHP](https://www.php.net/manual/en/function.password-hash.php)

We can avoid the issue with the `LongPasswordStrategies.none` strategy, but as we seem to respect the standard, I think we shouldn't need it, and that the package would benefit increasing the max length by 1.

What do you think?

Thanks.
Matt'.

cc @Indigo744 because I think you worked a lot on this topic.

**Comment by quinot** [comment]
Proposed PR: #44

**Comment by patrickfav** [comment]
Very fair point. Ive merged the PR and will see that I release a new version soon. Thank you!

**Comment by patrickfav** [comment]
Released with 0.10.0+
